crates/services/txpool_v2/src/service.rs

```diff
@@ -662,10 +674,20 @@ where
         let removed;
         {
             let mut pool = self.pool.write();
-            removed = pool.remove_transaction_and_dependents(txs_to_remove);
+            removed = pool.remove_transaction_and_dependents(txs_to_remove.into_iter());
         }
 
         for tx in removed {
+            {
+                let mut height_expiration_txs = self.pruner.height_expiration_txs.write();
+                let expiration = tx.expiration();
+                if expiration < u32::MAX.into() {
+                    if let Some(expired_txs) = height_expiration_txs.get_mut(&expiration)
+                    {
+                        expired_txs.remove(&tx.id());
+                    }
+                }
+            }
```

Comment:
Do you think it's worth to have a test for this additional behavior?

Reply:
It would be cool that all tests verify this state at the end and so the existing test will enforce this is working but this require to add a lot of ultis in test-helpers etc so I opened a followup issue : https://github.com/FuelLabs/fuel-core/issues/2581